src/mpid/ch4/netmod/ofi/util.c

```diff
@@ -7,12 +7,20 @@
 #include "ofi_impl.h"
 #include "ofi_events.h"
 
-int MPIDI_OFI_retry_progress(void)
+int MPIDI_OFI_retry_progress(int vci, int retry)
 {
     /* We do not call progress on hooks form netmod level
      * because it is not reentrant safe.
      */
-    return MPID_Progress_test(NULL);
+    int mpi_errno;
+    if (abs(retry) % 1000 == 0) {
```

Comment:
Should we use the same logic as `MPIDI_progress_test_vci`? E.g.
```c
    if (!MPIDI_VCI_IS_EXPLICIT(vci) && MPIDI_do_global_progress()) {
```
It might be nice to use that function directly by disabling SHM progress with another parameter, but not sure it's worth the churn.

Reply:
Spin progress to resolve a libfabric `EAGAIN` issue is quite specific. I am not sure they share the same concerns as the general MPI progress. I think we would prefer to treat these two progress separately so when the user adjust one CVAR, it won't affect the behavior of another.

Reply:
OK, fair enough. My initial thought was to just use a power of 2 for the frequency so the code generation for the modulo is a little cleaner. `MPIDI_do_global_progress` uses `global_vci_poll_count & MPIDI_CH4_PROG_POLL_MASK` for a fast comparison, which we could also do a version of here.

Reply:
I see. Makes sense. Will update